enhancements/disaster-recovery/automated-cluster-backups.md

```diff
@@ -0,0 +1,319 @@
+
+## Summary
+
+Enable the configuration of automated cluster backups.
+
+## Motivation
+
+The current documented procedure for performing a backup of an OpenShift
+cluster is manually initiated. This reduces the likelyhood of a timely backup
+being available in the event of catastrophic cluster failure.
+
+The procedure also requires gaining a root shell on a control plane
+node. Shell access to OpenShift control plane nodes access is generally
+discouraged due to the potential for affecting the reliability of the
+node. Indeed, at least one customer reports having had operator error result
+in a control plane node being taken offline in the process of attempting to
+follow the documented backup procedure.
+
+Finally, OpenShift 4.9 is intended to ship with etcd 3.5, and this will mean
+an upgrade for existing clusters from 3.4. Ensuring support for automated
+backups in the 4.8 edge required to upgrade to 4.9 will increase the
+likelyhood that customers will have cluster backups available in case of any
+trouble with the etcd upgrade.
+
+### Goals
+
+1. One-time cluster backup can be initiated without requiring a root shell.
+1. Automated backups can be configured as a day 2 operation.
+1. Cluster backups are written to the host filesystem of control plane nodes.
+
+### Non-Goals
+
+1. Backup to cluster (i.e. PVC) or cloud storage (e.g. S3).
+1. Support day 1 configuration of automated backups.
+1. Automate cluster restoration.
+1. Rewrite the existing cluster backup script.
+
+## Proposal
+
+- Add a new API singleton supporting backup configuration
+  - Suggested resource name `backups.config.openshift.io`
+    - Separate backup configuration from the etcd operator in case
+      implementation migrates to a dedicated backup operator.
+  - Fields:
+    - spec.reason `string`
+      - Set to a value different from the most recent reason, will prompt a
+        one-time backup.
+      - Default: `""`
+    - spec.schedule `string`
+      - [Standard cron expression](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax)
+      - Results in backups scheduled repeatedly to successive control plane
+        nodes.
+      - Default: `""`
+    - spec.maxDurationWithoutBackup `duration`
+      - If a backup has not successfully completed in the configured
+        interval, a critical alert will be generated for this and every
+        subsequent interval. The alert will either indicate that backups are
+        not configured, or supply the most recent reason a backup failed.
+      - Setting to `0` will disable alerting.
+      - Default: `1h`
+    - spec.retentionCount `int`
+      - The maximum number of backups to retain.
+      - If the number of succesful backups (as determined by recorded
+        status) matches `retentionCount`, the oldest backup will removed
+        before a new backup is initiated.
+      - Default: 15
```

Comment:
It's a completely arbitrary number. In the pantheon of things to worry about in this enhancement the default values are bottom of the list and it may make more sense to worry about them in the context of implementation PRs.